app/javascript/lib/validation/io.js

```diff
@@ -20,8 +22,49 @@ export async function readFileBytes(file, startByte=0, chunkSize=DEFAULT_CHUNK_S
   const blob = file.slice(startByte, nextSlice)
 
   const arrayBuffer = await blob.arrayBuffer()
-  const enc = new TextDecoder('utf-8')
 
+  const enc = new TextDecoder('utf-8')
   const stringContent = enc.decode(arrayBuffer)
+
+  return stringContent
+}
+
+/** Read an entire gzipped file, return string */
+export async function readGzipFile(file) {
+  const arrayBuffer = await file.arrayBuffer()
+  // Examples: https://github.com/101arrowz/fflate#usage
+  // See also:
+  // - https://github.com/101arrowz/fflate/discussions/30
+  // - https://github.com/101arrowz/fflate/discussions/119
+  //
+  // Consider refactoring ChunkedLineReader to enable real streaming-gunzip,
+  // to mitigate UI freezes for non-tiny gzipped files.  This might
+  // make sense to do as part of a larger refactor to parallelize
+  // byte-upload and file validation.
+  const uint8Array = new Uint8Array(arrayBuffer)
+
+  // const stringContent = strFromU8(gunzipSync(uint8Array))
```

Comment:
did you mean to leave in this commented-out line?

Reply:
Yes, but I've refined it with some commentary to clarify why it's here.